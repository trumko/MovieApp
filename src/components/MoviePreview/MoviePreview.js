import React from "react";
import { Link } from 'react-router-dom';

import { getMovieYear } from 'utils/helpers';

import './MoviePreview.scss';

export default function MoviePreview({ title, date, genres, posterUrl, id }) {
  const movieYear = getMovieYear(date);
  const movieGenres = genres.join(', ');
  return (
    <Link to={`/movie/${id}`} className="mr_moviePreview">
      <img src={posterUrl} className="mr_moviePoster" alt={title} />
      <div className="mr_movieInfo">
        <div>
          <div className="mr_movieTitle">{title}</div>
          <div className="mr_movieDate">
            <span>{movieYear}</span>
          </div>
        </div>
        <div className="mr_movieGenre">{movieGenres}</div>
      </div>
    </Link>
  );
}
