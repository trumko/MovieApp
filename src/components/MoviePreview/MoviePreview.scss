@import "styles/variables.scss";

.mr_moviePreview {
  width: 200px;
  display: inline-block;
  padding: 10px;

  &:hover {
    background: $greyLight;
  }

  &:hover img {
    filter: brightness(120%);
  }

  .mr_moviePoster {
    width: 100%;
  }

  .mr_movieInfo {
    padding: 5px 0;
  }

  .mr_movieTitle {
    text-transform: uppercase;
    font-weight: bold;
    font-size: 12px;
    margin-bottom: 5px;
    width: 80%;
    display: inline-block;
  }

  .mr_movieDate {
    width: 20%;
    display: inline-block;
    text-align: right;
    font-weight: bold;
    font-size: 12px;
    vertical-align: top;
  }

  .mr_movieDate span {
    text-align: right;
    border: 1px solid $greyDark;
    padding: 0 3px;
    border-radius: 2px;
  }

  .mr_movieGenre {
    color: $greyDark;
    font-weight: bold;
    font-size: 12px;
  }
}
