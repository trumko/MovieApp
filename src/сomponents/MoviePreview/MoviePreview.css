.mr_moviePreview {
  width: 195px;
  display: inline-block;
  padding: 10px;
}

.mr_moviePreview:hover {
  background: #e8e8e8;
}

.mr_moviePreview:hover img {
  filter: brightness(120%);
}


.mr_moviePoster {
  width: 100%;
}

.mr_movieInfo {
  padding: 5px;
}

.mr_movieTitle {
  text-transform: uppercase;
  font-weight: bold;
  font-size: 12px;
  margin-bottom: 5px;
  width: 80%;
  display: inline-block;
}

.mr_movieDate {
  width: 20%;
  display: inline-block;
  text-align: right;
  font-weight: bold;
  font-size: 12px;
}

.mr_movieDate span {
  text-align: right;
  border: 1px solid #757575;
  padding: 0 3px;
  border-radius: 2px;
}

.mr_movieGenre {
  color: #757575;
  font-weight: bold;
  font-size: 12px;
}
