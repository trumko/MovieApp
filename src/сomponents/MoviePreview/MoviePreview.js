import React from "react";
import { Link } from 'react-router-dom';

import './MoviePreview.css';
import inBrugesPoster from 'assets/in_bruges_poster.jpg';

export default function MoviePreview({ title, date, genre, posterUrl }) {
  return (
    <Link to="/" className="mr_moviePreview">
      <img src={posterUrl} className="mr_moviePoster" />
      <div className="mr_movieInfo">
        <div>
          <div className="mr_movieTitle">{title}</div>
          <div className="mr_movieDate">
            <span>{date}</span>
          </div>
        </div>
        <div className="mr_movieGenre">{genre}</div>
      </div>
    </Link>
  );
}
