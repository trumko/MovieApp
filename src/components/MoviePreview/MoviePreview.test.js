// import React from "react";
// import { Link } from 'react-router-dom';
//
// import { getMovieYear } from 'utils/helpers';
//
// import './MoviePreview.scss';
//
// export default function MoviePreview({ title, date, genres, posterUrl, id }) {
//   const movieYear = getMovieYear(date);
//   const movieGenres = genres.join(', ');
//   return (
//     <Link to={`/movie/${id}`} className="mr_moviePreview">
//       <img src={posterUrl} className="mr_moviePoster" alt={title} />
//       <div className="mr_movieInfo">
//         <div>
//           <div className="mr_movieTitle">{title}</div>
//           <div className="mr_movieDate">
//             <span>{movieYear}</span>
//           </div>
//         </div>
//         <div className="mr_movieGenre">{movieGenres}</div>
//       </div>
//     </Link>
//   );
// }
//

import React from 'react'
import { shallow } from 'enzyme'
import MoviePreview from './MoviePreview'

describe('MoviePreview', () => {
  it('MoviePreview with custom Movie should be rendered correctly', () => {
    const component = shallow(
      <MoviePreview
        title="Isle of Dogs"
        date="2018-03-23"
        genres={["Adventure", "Comedy", "Animation"]}
        posterUrl="https://image.tmdb.org/t/p/w500/c0nUX6Q1ZB0P2t1Jo6EeFSVnOGQ.jpg"
        id="399174"

      />
    );

    expect(component).toMatchSnapshot();
  });
});
