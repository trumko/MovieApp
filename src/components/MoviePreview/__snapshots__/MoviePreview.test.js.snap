// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePreview MoviePreview with custom Movie should be rendered correctly 1`] = `
<Link
  className="mr_moviePreview"
  to="/movie/399174"
>
  <img
    alt="Isle of Dogs"
    className="mr_moviePoster"
    src="https://image.tmdb.org/t/p/w500/c0nUX6Q1ZB0P2t1Jo6EeFSVnOGQ.jpg"
  />
  <div
    className="mr_movieInfo"
  >
    <div>
      <div
        className="mr_movieTitle"
      >
        Isle of Dogs
      </div>
      <div
        className="mr_movieDate"
      >
        <span>
          2018
        </span>
      </div>
    </div>
    <div
      className="mr_movieGenre"
    >
      Adventure, Comedy, Animation
    </div>
  </div>
</Link>
`;
